plugins/org.jboss.tools.openshift.express.ui/src/org/jboss/tools/openshift/express/internal/ui/explorer/ExplorerExpressContentProvider.java

```diff
@@ -66,10 +64,16 @@ public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
 		loadedElements.clear();
 		loadingElements.clear();
 		errors.clear();
-		if (parentElement instanceof IWorkspaceRoot) {
-			return ConnectionsRegistrySingleton.getInstance().getAll();
-		} else if (parentElement instanceof ConnectionsRegistry) {
-			return ((ConnectionsRegistry) parentElement).getAll();
+		if (parentElement instanceof ConnectionsRegistry) {
+			ConnectionsRegistry registry = (ConnectionsRegistry) parentElement;
+			IConnection[] all = registry.getAll();
+			List<IConnection> conns = new ArrayList<IConnection>(all.length);
```

Comment:
Furthermore I think that #getAll() should return a Collection<IConnection> (it currently returns an array IConnection[]) like #get(Class<T> clazz) currently does.